yarl/_path.py

```diff
@@ -39,3 +40,24 @@ def normalize_path(path: str) -> str:
 
     segments = path.split("/")
     return prefix + "/".join(normalize_path_segments(segments))
+
+
+def relative_path(path: str, start: str) -> str:
+    """A wrapper over os.path.relpath()"""
+
+    if not path:
+        path = "/"
+    if not start:
+        start = "/"
+    if not start.endswith("/"):
+        start = dirname(start)
+
+    if (path.startswith("/") and not start.startswith("/")) or (
+        not path.startswith("/") and start.startswith("/")
+    ):
```

Comment:
Let's try to explain the semantics of the check by labeling them:
```suggestion
    path_has_leading_slash = path.startswith("/")
    start_has_leading_slash = start.startswith("/")
    both_have_leading_slash = all((path_has_leading_slash, start_has_leading_slash))
    none_have_leading_slash = all((not path_has_leading_slash, not start_has_leading_slash))
    if not both_have_leading_slash and not none_have_leading_slash:
```

Reply:
@oleksbabieiev pro tip: it's usually recommended to accept code suggestions via GH UI, since that makes sure there's no accidental partial copy-paste and preserves the credit.